src/Tizen.BatteryMonitor/Tizen.BatteryMonitor/BatteryMonitor.cs

```diff
@@ -0,0 +1,206 @@
+            if (((BatteryMonitorError)ret != BatteryMonitorError.None))
+            {
+                Log.Error(Globals.LogTag, "Error getting battery usage for all resources" + (BatteryMonitorError)ret);
+                throw BatteryMonitorErrorFactory.ThrowBatteryMonitorException(ret);
+            }
+
+            foreach(ResourceType type in rtypes) {
+                batteryUsage.Add(_instance.GetPowerUsageForResource(bmHandle,type));
+            }
+            ret = Interop.BatteryMonitor.Destroy(bmHandle);
+            if (((BatteryMonitorError)ret != BatteryMonitorError.None))
+            {
+                Log.Error(Globals.LogTag, "Error in Destroy handle, " + (BatteryMonitorError)ret);
+                throw BatteryMonitorErrorFactory.ThrowBatteryMonitorException(ret);
+            }
+            return batteryUsage;
+        }
+
+        /// <summary>
+        /// Gets the battery consumption in mAh(milli-Ampere hour) for the specific resource for the given application in custom interval.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="rtype">Identifier of resource type. BLE, WiFi, CPU etc.</param>
+        /// <param name="stime">Start Time for data in Epoch Time (in seconds).</param>
+        /// <param name="etime">End Time for data in Epoch Time (in seconds).</param>
+        /// <returns>Returns the battery consumption in mAh(milli-Ampere hour) for the specific resource for the given application in custom interval.</returns>
+        public double GetPowerUsageByAppForResource(string appID, ResourceType rtype, long stime, long etime)
+        {
+            double batteryUsage = 0;
+            int ret = Interop.BatteryMonitor.GetPowerUsageByAppForResource(appID, rtype, stime, etime, out batteryUsage);
+            if (((BatteryMonitorError)ret != BatteryMonitorError.None))
+            {
+                Log.Error(Globals.LogTag, "Error getting battery usage by app for resrource" + (BatteryMonitorError)ret);
+                throw BatteryMonitorErrorFactory.ThrowBatteryMonitorException(ret);
+            }
+            return batteryUsage;
+        }
+
+        /// <summary>
+        /// Gets the total battery usage in mAh(milli-Ampere hour) by an application for certain time interval.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="stime">Start Time for data in Epoch Time (in seconds).</param>
+        /// <param name="etime">End Time for data in Epoch Time (in seconds).</param>
+        /// <returns>Returns the total battery usage in mAh(milli-Ampere hour) by an application for certain time interval.</returns>
+        public double GetTotalPowerUsageByApp(string appID, long stime, long etime)
```

Comment:
Use standard type on date https://docs.microsoft.com/ko-kr/dotnet/api/system.datetime?view=netframework-4.8
```suggestion
        public double GetTotalPowerUsageByApp(string appID, DateTime start, DateTime end)
```